quesma/schema/registry.go

```diff
@@ -5,21 +5,29 @@ package schema
 import (
 	"quesma/logger"
 	"quesma/quesma/config"
-	"strings"
+	"quesma/util"
+	"sync"
 )
 
 type (
 	Registry interface {
 		AllSchemas() map[TableName]Schema
 		FindSchema(name TableName) (Schema, bool)
 		UpdateDynamicConfiguration(name TableName, table Table)
+		UpdateFieldEncodings(encodings map[FieldEncodingKey]string)
+	}
+	FieldEncodingKey struct {
+		TableName string
+		FieldName string
 	}
 	schemaRegistry struct {
 		// index configuration overrides always take precedence
 		indexConfiguration      *map[string]config.IndexConfiguration
 		dataSourceTableProvider TableProvider
 		dataSourceTypeAdapter   typeAdapter
 		dynamicConfiguration    map[string]Table
+		fieldEncodingsLock      sync.RWMutex
+		fieldEncodings          map[FieldEncodingKey]string
```

Comment:
additionally to what I've wrote about this belonging to `Schema` and removing the redundant reference to `table name` - I think we should make this `map[FieldEncodingKey]InternalFieldName` or maybe `map[FieldEncodingKey]EncodedFieldName` so just adding a meaningful type alias :)